### PR TITLE
docs: remove unneeded @throws in CodeIgniter

### LINKED ISSUE #7434: Bug: PhpStorm warnings in public/index.php
### PHP Version

8.2

### CodeIgniter4 Version

4.3.3 updated today

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

MariaDB 10.4.28

### What happened?

phpStorm throwing Warnings on these 2 lines now.

Line #16 - Can be replaced with 'const' syntax

// Path to the front controller (this file)
define('FCPATH', )__DIR__ . DIRECTORY_SEPARATOR;

Line #67 - Unhandled \CodeIgniter\Router\Exceptions\RedirectException
$app->run();

### Steps to Reproduce

Open in phpStorm
Top right Warning sign.

### Expected Output

No Warnings.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Closes #7434
See https://github.com/codeigniter4/CodeIgniter4/issues/7434#issuecomment-1507972531

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/CodeIgniter.php [hunk 1]
```diff
@@ -310,8 +310,6 @@ private function configureKint(): void
      * makes all of the pieces work together.
      *
      * @return ResponseInterface|void
-     *
-     * @throws RedirectException
```
Comment (samsonasik) [inline system/CodeIgniter.php:314]: since the method has `LogicException` throw, how about change to `@throws LogicException` ?

Comment (kenjis) [reply system/CodeIgniter.php:314]: I think we should write Exceptions in `@throws` which should be caught in the client code.
LogicException should not be caught.

Comment (paulbalandan) [reply system/CodeIgniter.php:314]: I think adding `@throws` doesn't mean it have to be caught in client code. It's a way of explicitly telling that calling the function may throw this exception.

Comment (kenjis) [reply system/CodeIgniter.php:314]: @paulbalandan Do you think we should write all Exceptions that may be thrown?

Comment (paulbalandan) [reply system/CodeIgniter.php:314]: Not really *all*. Only those that are meant to be checked by the framework and/or user. Please read this [PHPStan article](https://phpstan.org/blog/bring-your-exceptions-under-control) for details.

Comment (kenjis) [reply system/CodeIgniter.php:314]: I read the article.
I think checked exceptions should be caught. So we should write only checked exceptions in `@thrown`.
Something wrong with my understanding?

**Comment by iRedds** [comment]
You forgot to fix \CodeIgniter\Test\FeatureTestTrait and \CodeIgniter\Test\FeatureTestCase.

**Comment by kenjis** [comment]
@iRedds Thanks. Removed.

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**